### PR TITLE
Fix php warning on update item qty request from sidebar

### LINKED ISSUE #34380: Warning: A non-numeric value encountered in vendor/magento/module-checkout/Controller/Sidebar/UpdateItemQty.php on line 69
### Preconditions (*)

Magento 2.4.3

### Steps to reproduce (*)

1. Open https://domain.com/checkout/sidebar/updateItemQty/?item_qty=error
2. See 500 error

### Expected result (*)

No 500 error

### Actual result (*)

`Warning: A non-numeric value encountered in /vendor/magento/module-checkout/Controller/Sidebar/UpdateItemQty.php on line 69`

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by Melnychuk-Alexandr**
### Description (*)

Added "int" type when get param from request.
Path to file (app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php)

### Fixed Issues (if relevant)

1. Fixes magento/magento2#34380

### Manual testing scenarios (*)

1. Open  https://your-magento-domain.com/checkout/sidebar/updateItemQty/?item_qty=error
2. See result

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function __construct(
     public function execute()
     {
         $itemId = (int)$this->getRequest()->getParam('item_id');
-        $itemQty = $this->getRequest()->getParam('item_qty') * 1;
+        $itemQty = (int)$this->getRequest()->getParam('item_qty') * 1;
```
Comment (ihor-sviziev) [inline app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: I'm pretty sure qty might be float
```suggestion
        $itemQty = (float)$this->getRequest()->getParam('item_qty') * 1;
```

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please also add test coverage for this case

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php (hunk 1 above)
Comment (Den4ik) [reply app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: Totally agree with @ihor-sviziev

**Review by Den4ik — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php (hunk 1 above)
Comment (Melnychuk-Alexandr) [reply app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: Thanks for advice guys!

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @Melnychuk-Alexandr,
Could you please cover your change with any type of automated test?
Thank you!

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Could you also restrict this controller to handle only POST requests?
I see it's only used in https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/Checkout/view/frontend/templates/cart/form.phtml#L13-L20

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php [hunk 2]
```diff
@@ -62,11 +62,16 @@ public function __construct(
...
     /**
      * @return $this
+     * @throws LocalizedException
      */
     public function execute()
     {
+        if (!$this->getRequest()->isPost()) {
+            throw new LocalizedException(__('Wrong request.'));
+        }
+
```
Comment (ihor-sviziev) [inline app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: Please revert this change and add add to the class that it implements `Magento\Framework\App\Action\HttpPostActionInterface` interface
```suggestion
```

Also, please cover your change with any type of automated test

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php (hunk 2 above)
Comment (Den4ik) [reply app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php]: I believe that unt tests is enough for this case

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Melnychuk-Alexandr** [comment]
@magento run Static Tests, Functional Tests CE

**Comment by Melnychuk-Alexandr** [comment]
Hi @ihor-sviziev @Den4ik . Added to implementation "HttpPostActionInterface", for which case I need to write automated test ?

**Comment by ihor-sviziev** [comment]
@Melnychuk-Alexandr, the case described in the related issue - when the item_qty isn't a number

**Comment by Melnychuk-Alexandr** [comment]
@magento run all tests

**Comment by Melnychuk-Alexandr** [comment]
@magento run Static Tests, Functional Tests CE, Functional Tests B2B

**Comment by Melnychuk-Alexandr** [comment]
@magento run Static Tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests look not related to changes from this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9398](https://jira.corp.magento.com/browse/ENGCOM-9398) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Alfa** [comment]
https://github.com/magento run Functional Tests B2B

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**

1. Have Magento latest installed
2. Make sure to have one customer login to login at the frontend


**Manual testing scenario:**

1. Login to the front end

2. Add a product to the cart and checkout.
 
3. Open developer tool and be in the Network tab;

4. Replace the URL to this - https://your-magento-domain.com/checkout/sidebar/updateItemQty/?item_qty=error and enter


**Before:** :heavy_multiplication_x: We were getting an 500 error in the response

image


**After:** :heavy_check_mark: We are getting an Magento error with 404

image



There is no additional testing required since it is specific to URL tampering, hence no regression case has been added for this.

**Comment by engcom-Alfa** [comment]
https://github.com/magento run Functional Tests B2B

**Comment by engcom-Alfa** [comment]
@magento run Functional Tests B2B

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9398](https://jira.corp.magento.com/browse/ENGCOM-9398) has been created to process this Pull Request

**Comment by engcom-Charlie** [comment]
As PR has been approved and tested, also the build is green now, hence moving this PR to Merge in Progress.

Thank you!

**Comment by davidandersson1** [comment]
Hello,

Anybody know how to apply the patch for this issue?

I have downloaded https://github.com/magento/magento2/commit/b692ea60a10f695cfb3114c5faf6b8dd60548b32.diff
and created folder at "app/code/Magento/Checkout/Controller/Sidebar" and put file UpdateItemQty.php as per this
https://github.com/Melnychuk-Alexandr/magento2/blob/b692ea60a10f695cfb3114c5faf6b8dd60548b32/app/code/Magento/Checkout/Controller/Sidebar/UpdateItemQty.php

and tried with composer i.e
"magento/module-checkout": {
                "non-numeric value encountered": "patches/composer/b692ea60a10f695cfb3114c5faf6b8dd60548b32.diff",              
            },
but somehow it's not work and throwing error as per attached screenshot.

So please help if anybody know exactly how to apply this patch.
Screenshot from 2022-08-02 11-51-23

**Comment by hostep** [comment]
@davidandersson1:

1. overriding a file by just putting it in `app/code` is most likely not going to work, this isn't Magento 1 anymore.
2. The patch you need is not the first commit of this PR, but all the commits of the PR, in this case: https://github.com/magento/magento2/pull/34953.diff
3. If the patch doesn't apply, it could mean that it's not compatible with your magento version (but you are on 2.4.3-p1 or 2.4.3-p2, and the patch should apply on those versions), or it could mean that you haven't updated the correct paths in your patch, so try to understand how your patches module works by reading its documentation. Also adding `-vvv` to `composer install` might show you more details about why it fails.

**Comment by davidandersson1** [comment]
@hostep thanks for your response, I took all commits of the PR and apply patch accordingdly and now patch works fine. thanks a lot.

**Pull request merged**